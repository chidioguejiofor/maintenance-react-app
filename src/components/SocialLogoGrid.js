/* eslint react/no-array-index-key: off */
import React from 'react';

const social = [
  {
    name: 'facebook',
    link: '#',
    imgSrc: 'https://image.flaticon.com/icons/svg/145/145802.svg',
    width: '30px',
  },
  {
    name: 'linkedin',
    link: '#',
    imgSrc: 'https://image.flaticon.com/icons/svg/145/145807.svg',
    width: '30px',
  },
  {
    name: 'twitter',
    link: '#',
    imgSrc: 'https://image.flaticon.com/icons/svg/145/145812.svg',
    width: '30px',
  },
  {
    name: 'google',
    link: '#',
    imgSrc: 'https://image.flaticon.com/icons/svg/145/145804.svg',
    width: '30px',
  },
];
const SocialLogoGrid = (() => (
  <ul className=" horizontal-list flex-center-box">
    {social instanceof Array
    && social.map(({ link, imgSrc, name, width, height }, index) => (
      <li key={index}>
        <a href={link}>
          <img
            src={imgSrc}
            alt={name}

            width={width || 'auto'}
            height={height || 'auto'}
          />
        </a>
      </li>
    ))}
  </ul>
)
);


export default SocialLogoGrid;
