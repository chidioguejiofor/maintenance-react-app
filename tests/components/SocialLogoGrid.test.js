import React from 'react';
import { shallow } from 'enzyme';

import SocialLogoGrid from '../../src/components/SocialLogoGrid';

const socialMedias = [
  {
    name: 'firstSocial',
    link: 'firstLink',
    imgSrc: 'facebook.png',
  },
  {
    name: 'secondSocial',
    link: 'secondLink',
    imgSrc: 'secondSocial.png',
  },
];


describe('Testing social logo grid', () => {
  it('should match snapshot', () => {
    expect(shallow(<SocialLogoGrid />)).toMatchSnapshot();
  });
});
