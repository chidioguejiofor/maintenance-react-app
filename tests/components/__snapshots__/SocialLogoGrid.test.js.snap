// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing social logo grid should match snapshot 1`] = `
<ul
  className=" horizontal-list flex-center-box"
>
  <li
    key="0"
  >
    <a
      href="#"
    >
      <img
        alt="facebook"
        height="auto"
        src="https://image.flaticon.com/icons/svg/145/145802.svg"
        width="30px"
      />
    </a>
  </li>
  <li
    key="1"
  >
    <a
      href="#"
    >
      <img
        alt="linkedin"
        height="auto"
        src="https://image.flaticon.com/icons/svg/145/145807.svg"
        width="30px"
      />
    </a>
  </li>
  <li
    key="2"
  >
    <a
      href="#"
    >
      <img
        alt="twitter"
        height="auto"
        src="https://image.flaticon.com/icons/svg/145/145812.svg"
        width="30px"
      />
    </a>
  </li>
  <li
    key="3"
  >
    <a
      href="#"
    >
      <img
        alt="google"
        height="auto"
        src="https://image.flaticon.com/icons/svg/145/145804.svg"
        width="30px"
      />
    </a>
  </li>
</ul>
`;
